### PR TITLE
Implement 3 level namespace

### PR DESCRIPTION
**Pull request by bogdanghit**
This patch adds support for a 3-level namespace such as `catalog`.`schema`.`table`.

### CODE REVIEW

File: tableau-databricks/manifest.xml [hunk 1]
```diff
@@ -17,6 +17,11 @@ limitations under the License.
 -->
 
 <connector-plugin class='databricks' superclass='spark' plugin-version='1.0' name='@string/databricks/' version='18.1'>
+<vendor-information>
+    <company name="Test Company"/>
+    <support-link url="https://example.com"/>
+    <driver-download-link url="https://drivers.example.com"/>
+</vendor-information>
```
Comment (bogdanghit) [inline tableau-databricks/manifest.xml:24]: remove this before shipping

File: tableau-databricks/connection-metadata.xml [hunk 2]
```diff
@@ -16,9 +16,7 @@ See the License for the specific language governing permissions and
 limitations under the License. -->
 
 <connection-metadata>
-    <database enabled='true' label="@string/http_path_prompt/" >
-        <field optional='false' />
-    </database>
+    <database enabled='true' label="@string/metadata_catalog_prompt/" />
```
Comment (bogdanghit) [inline tableau-databricks/connection-metadata.xml:19]: investigate what label do we need here to show up `Catalog`

File: tableau-databricks/connection-builder.js [hunk 3]
```diff
@@ -18,10 +18,14 @@ limitations under the License.
 (function dsbuilder(attr) {
 	var params = {};
 
+  var vendorDefined = {	
+		attributeHttpPath: "v-http-path"	
+	}
```
Comment (juliuszsompolski) [inline tableau-databricks/connection-builder.js:23]: Do we need Tableau help for the backwards compat?
Can we just write here
```
if (connectionHelper.attributeDatabase is set AND vendorDefined.attributeHttpPath is not set) {
  // for old workbooks
  params["HTTPPATH"] = attr[connectionHelper.attributeDatabase];
} else if (connectionHelper.attributeDatabase is not set AND vendorDefined.attributeHttpPath is set) {
  // for new workbooks
  params["HTTPPATH"] = attr[vendorDefined.attributeHttpPath];
} else if (connectionHelper.attributeDatabase is set AND vendorDefined.attributeHttpPath is set) {
  // for new workbooks with multiple catalogs
  params["<CATALOG>"] = connectionHelper.attributeDatabase;
  params["HTTPPATH"] = attr[vendorDefined.attributeHttpPath];
}

Comment (juliuszsompolski) [reply tableau-databricks/connection-builder.js:23]: Actually, scrap the last `else if`. You told me that when dbname is not specified, it just uses GetCatalogs and we're good.

Comment (CJStuart) [reply tableau-databricks/connection-builder.js:23]: IIRC the problem occurred before the JS invocation, but I'm not sure if I'm remembering that correctly.

File: tableau-databricks/manifest.xml (hunk 1 above)
Comment (bogdanghit) [reply tableau-databricks/manifest.xml:24]: Don't forget to remove vendor info.

File: tableau-databricks/connection-fields.xml [hunk 4]
```diff
@@ -23,6 +23,10 @@ limitations under the License. -->
     </field>
     <!-- /Endpoint -->
 
+    <field name="v-http-path" label="@string/http_path_prompt/" category="endpoint" value-type="string">	
+        <!-- validation-rule reg-exp="^[0-9A-Za-z-_/]*$" /-->
```
Comment (bogdanghit) [inline tableau-databricks/connection-fields.xml:27]: we can keep this line, but the indentation is wrong

Comment (alismess-db) [reply tableau-databricks/connection-fields.xml:27]: Have we decided that want to enable verification here or not?

Comment (alismess-db) [reply tableau-databricks/connection-fields.xml:27]: Similarly concerning the Server Hostname validation rule: if a company has configured "mycompanyname-databricks.com" to redirect to "adb-94384578309420.databricks.com", then using the first URL would not work in Tableau. Do you think the validation is needlessly restrictive?

Comment (alismess-db) [reply tableau-databricks/connection-fields.xml:27]: Or actually, never mind about the Server Hostname validation. I misread the regex.

Comment (bogdanghit) [reply tableau-databricks/connection-fields.xml:27]: I don't remember why I disabled http path validation here. Does it break if it's enabled?

Comment (bogdanghit) [reply tableau-databricks/connection-fields.xml:27]: Oh wait, actually it was disabled before my change.

Comment (alismess-db) [reply tableau-databricks/connection-fields.xml:27]: HTTP Path wasn't validated previously as it was simply the `<database>` field.

**Review by bogdanghit — COMMENTED** [review]
LGTM

**Pull request closed without merging**